### PR TITLE
Start to drop Python 2.7

### PR DESCRIPTION
**Pull request by joshuagl**
Fixes: #

### Description of the changes being introduced by the pull request:

This is a first PR towards removing Python 2.7 from the code base, it is not sufficient but it does remove enough pieces that people will not be tempted to think we are still supporting Python 2.7 **and** starts getting us newer versions of cryptography via dependabot.

For more detail on what else needs to be removed, see https://github.com/secure-systems-lab/securesystemslib/issues/344

### Please verify and check that the pull request fulfils the following requirements:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -103,7 +101,7 @@
     'Source': 'https://github.com/secure-systems-lab/securesystemslib',
     'Issues': 'https://github.com/secure-systems-lab/securesystemslib/issues',
   },
-  python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*, <4",
+  python_requires="!=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*, <4",
```
Comment (jku) [inline setup.py:104]:
```suggestion
  python_requires="~=3.6",
```
I think the "compatible release" clause that TUF uses works here too (that should translate to any 3.X version where X>=6)

Comment (joshuagl) [reply setup.py:104]: Yes, thank you. I even introduced that clause in tuf 🤦

**Review by jku — APPROVED** [review]
lgtm: there's still mentions of python2 even in the same files but this seems a fine first step

**Comment by joshuagl** [comment]
> lgtm: there's still mentions of python2 even in the same files but this seems a fine first step

Thanks. I've pushed an extra commit to drop Python2 dependencies and dependency pinning (9b3a513).

**Pull request merged**